sbi/neural_nets/embedding_nets/SC_embedding.py

```diff
@@ -0,0 +1,311 @@
+        """
+
+        # Compute Fourier coeffcients up to factor of e^(- something constant)
+        x_ft = transform.forward(x.cfloat(), norm='forward')
+
+        x_ft = x_ft.permute(0, 2, 1)
+        x_ft = self.compl_mul1d(x_ft, self.weights1)  # (batch, conv_channels, modes)
+        x_ft = x_ft.permute(0, 2, 1)  # (batch, modes, conv_channels)
+        x_ft = torch.view_as_real(x_ft)  # (batch, modes, conv_channels, 2)
+        x_ft = x_ft.permute(0, 1, 3, 2)
+        x_ft = x_ft.reshape(x.shape[0], 2 * self.modes, self.out_channels)
+
+        return x_ft
+
+
+class SpectralConvEmbedding(nn.Module):
+    def __init__(
+        self,
+        in_channels: int,
+        modes: int = 20,
+        out_channels: int = 1,
+        conv_channels: int = 5,
+        num_layers: int = 3,
+    ):
+        """SpectralConvEmbedding is a neural network module that performs convolution
+        in Fourier space for 1D input data (that can have multiple channels).
+        It uses a series of spectral convolution layers and pointwise
+        convolution layers to transform the input tensor.
+
+        Args:
+            in_channels: Number of channels in the input data.
+            modes: Number of modes considered in the spectral convolution,
+                at most floor(n_points/2) + 1.
+            out_channels: number of channels for final output.
+            conv_channels: Number of going in and out convolutional layer.
+            num_layers: Number of convolution layers.
+
+        """
+        super().__init__()
+
+        self.modes = modes
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.conv_channels = conv_channels
+        self.num_layers = num_layers
+
+        # Initialize fully connected layer to raise number of
+        # input channels to number of convolutional channels
+        self.fc0 = nn.Linear(self.in_channels, self.conv_channels)
+
+        # Inititalize layers performing convolution in Fourier space
+        self.conv_layers = nn.ModuleList([
+            SpectralConv1d_SMM(self.conv_channels, self.conv_channels, self.modes)
+            for _ in range(self.num_layers)
+        ])
+
+        # Initialize layer performing pointwise convolution
+        self.w_layers = nn.ModuleList([
+            nn.Conv1d(self.conv_channels, self.conv_channels, 1)
+            for _ in range(self.num_layers)
+        ])
+
+        # Initialize last convolutional layer with output in Fourier space
+        self.conv_last = SpectralConv1d_SMM(
+            self.conv_channels, self.conv_channels, self.modes
+        )
+
+        # Initialize fully connected layer to reduce number of output channels
+        self.fc_last = nn.Linear(self.conv_channels, self.out_channels)
+
+    def forward(self, x: Tensor) -> Tensor:
+        """Network forward pass.
+
+        Args:
+            x: 2D input tensor (batch_size, channels * n_points) for equi-spaced data or
```

Comment:
Thanks for the thorough docstring! Maybe it would be good to provide a code snippet example here of the two types of input ,i.e. how we expect to use and call `SpectralConvEmbedding`

Reply:
Also, I think that this is a bit weird to expect to pass `x` as `(batch_size, channels*n_points)`. It feels a lot more natural to pass `x` as `(batch_size,num_channels,n_points)`.